diff -ruN musl-1.2.3/src/errno/__errno_location.c musl-1.2.3-mine/src/errno/__errno_location.c
--- musl-1.2.3/src/errno/__errno_location.c	2021-01-15 03:26:00.000000000 +0100
+++ musl-1.2.3-mine/src/errno/__errno_location.c	2021-07-15 15:25:43.635856910 +0200
@@ -1,9 +1,11 @@
 #include <errno.h>
 #include "pthread_impl.h"
 
+int errno_global = 0;
+
 int *__errno_location(void)
 {
-	return &__pthread_self()->errno_val;
+  return &errno_global;
 }
 
 weak_alias(__errno_location, ___errno_location);
