diff -ruN musl-1.2.2/src/env/__init_tls.c musl-1.2.2-mine/src/env/__init_tls.c
--- musl-1.2.2/src/env/__init_tls.c	2021-01-15 03:26:00.000000000 +0100
+++ musl-1.2.2-mine/src/env/__init_tls.c	2021-07-01 14:48:40.607938601 +0200
@@ -87,6 +87,9 @@
 	size_t base = 0;
 	void *mem;

+  /* HACK : disable TLS */
+  return;
+
 	for (p=(void *)aux[AT_PHDR],n=aux[AT_PHNUM]; n; n--,p+=aux[AT_PHENT]) {
 		phdr = (void *)p;
 		if (phdr->p_type == PT_PHDR)
diff -ruN musl-1.2.2/src/env/__reset_tls.c musl-1.2.2-mine/src/env/__reset_tls.c
--- musl-1.2.2/src/env/__reset_tls.c	2021-01-15 03:26:00.000000000 +0100
+++ musl-1.2.2-mine/src/env/__reset_tls.c	2021-07-01 14:48:51.332004924 +0200
@@ -4,6 +4,9 @@

 void __reset_tls()
 {
+  /* HACK : disable TLS */
+  return;
+
 	pthread_t self = __pthread_self();
 	struct tls_module *p;
 	size_t i, n = self->dtv[0];
