diff -ruN ncurses-6.3/misc/gen-pkgconfig.in ncurses-6.3-mine/misc/gen-pkgconfig.in
--- ncurses-6.3/misc/gen-pkgconfig.in	2021-08-07 23:36:33.000000000 +0200
+++ ncurses-6.3-mine/misc/gen-pkgconfig.in	2022-07-25 20:43:09.275790602 +0200
@@ -50,7 +50,7 @@
 CXX_LIB_SUFFIX=@CXX_LIB_SUFFIX@
 
 suffix=@PC_MODULE_SUFFIX@
-prefix="@prefix@"
+prefix="@DESTDIR@"
 exec_prefix="@exec_prefix@"
 includedir="@includedir@@includesubdir@"
 libdir="@libdir@"
@@ -58,7 +58,7 @@
 RPATH_LIST=@RPATH_LIST@
 PRIVATE_LIBS="@PRIVATE_LIBS@"
 
-show_prefix='@prefix@'
+show_prefix='@DESTDIR@'
 show_exec_prefix='@exec_prefix@'
 show_includedir='@includedir@@includesubdir@'
 show_libdir='@libdir@'
