diff -ru musl-1.2.0-ori/arch/i386/syscall_arch.h musl-1.2.0-mine/arch/i386/syscall_arch.h
--- musl-1.2.0-ori/arch/i386/syscall_arch.h	2020-02-21 01:37:02.000000000 +0100
+++ musl-1.2.0-mine/arch/i386/syscall_arch.h	2021-05-09 19:02:14.180668381 +0200
@@ -3,11 +3,7 @@
 ((union { long long ll; long l[2]; }){ .ll = x }).l[1]
 #define __SYSCALL_LL_O(x) __SYSCALL_LL_E((x))
 
-#if SYSCALL_NO_TLS
 #define SYSCALL_INSNS "int $128"
-#else
-#define SYSCALL_INSNS "call *%%gs:16"
-#endif
 
 #define SYSCALL_INSNS_12 "xchg %%ebx,%%edx ; " SYSCALL_INSNS " ; xchg %%ebx,%%edx"
 #define SYSCALL_INSNS_34 "xchg %%ebx,%%edi ; " SYSCALL_INSNS " ; xchg %%ebx,%%edi"
